### ISSUE TITLE
yarn start gives error

### ISSUE THREAD

**Issue by b-zee** [maidsafe/safe_browser#118]
Operation System: Arch Linux
Beaker Version: master of safe_browser

Following the exact steps in the README leads to `yarn start` giving the following output:

```
Error: Dynamic Symbol Retrieval Error: /home/benno/safe_browser/app/node_modules/safe-app/src/native/libsafe_app.so: undefined symbol: test_create_app
    at DynamicLibrary.get (/home/benno/safe_browser/app/node_modules/ffi/lib/dynamic_library.js:112:11)
    at /home/benno/safe_browser/app/node_modules/safe-app/src/native/lib.js:27:42
    at Array.forEach (native)
    at Object.<anonymous> (/home/benno/safe_browser/app/node_modules/safe-app/src/native/lib.js:23:5)
    at Object.<anonymous> (/home/benno/safe_browser/app/node_modules/safe-app/src/native/lib.js:51:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Dynamic Symbol Retrieval Error: /home/benno/safe_browser/app/node_modules/safe-app/src/native/libsafe_app.so: undefined symbol: test_create_app
    at DynamicLibrary.get (/home/benno/safe_browser/app/node_modules/ffi/lib/dynamic_library.js:112:11)
    at /home/benno/safe_browser/app/node_modules/safe-app/src/native/lib.js:27:42
    at Array.forEach (native)
    at Object.<anonymous> (/home/benno/safe_browser/app/node_modules/safe-app/src/native/lib.js:23:5)
    at Object.<anonymous> (/home/benno/safe_browser/app/node_modules/safe-app/src/native/lib.js:51:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
```

This also happens with `yarn run watch`.

**Comment by shankar2105** [comment]
@b-zee When [safe-app](https://github.com/maidsafe/safe_app_nodejs) loads `libsafe_app` library, it dones't find `test_create_app` function in it. If you are using mock-routing, build the library with `testing` features with following command `cargo build --features "use-mock-routing testing"`.

**Comment by shankar2105** [comment]
@b-zee Just noticed that you are using Arc-Linux which we don't support for now. But give a try with above solution.

**Comment by mnoster** [comment]
I'm having the same issue on my Mac. Guess I'll give it a try on my Windows machine

**Comment by b-zee** [comment]
Where can I build the library manually? With NPM, NODE_ENV=dev had to be passed which apparently built the library with the mock router. Is there a way to provide yarn with options for the SAFE client library?

**Comment by shankar2105** [comment]
@b-zee remove `node_modules` and `app/node_module` folder, set `NODE_ENV=dev` and install the dependencies. Now try running the browser.

**Comment by mnoster** [comment]
Still encountering the issue for both npm and yarn. Please advise.

**Comment by bochaco** [comment]
@b-zee , @mnoster , can you please pull latest code from the Browser's master branch and try again as this should have been solved now. 
Please make sure you clean your `node_modules` and `app/node_modules` folders, and make sure you have the `NODE_ENV=dev` env variable set up for building. Then run `yarn install && yarn run rebuild && yarn run build`.

**Comment by b-zee** [comment]
@bochaco, it builds fine!

However, the browser freezes once I authorise the SAFE browser app. When I restart the browser, it's listed as authorised, but the overlay asks me to authorise it again. A test application seems to not be able to find any requested data. `window.safeApp.refreshContainersPermissions` for example.
Something seems off with the mock vault?

**Comment by bochaco** [comment]
Thanks @b-zee . We are aware of the other issues you mentioned, they issue with the browser hanging is tracked with https://maidsafe.atlassian.net/browse/MAID-2275, and the one about refreshing container access is believed to be due to the issue tracked with https://maidsafe.atlassian.net/browse/MAID-2262 which seems to be only with the the mock.
The overlay is being shown because you must have the lock option locked, so any app that requests access triggers the popup/overlay. Although, you are bringing up a good point, perhaps this case of the browser's app (which is the one that stores your bookmarks and history) must probably be an exception to the lock option rule. I'll raise this internally for consideration.

**Comment by b-zee** [comment]
I did not have the lock option locked, on opening the browser window it will ask me to authorise, no matter whether it's locked or not. My test application does not seem to get a response from `window.safeApp.authorise`. The application will not lead to an authorisation window pop-up.

I can not view the authorisation page of the SAFE browser app (which shows which permissions are given per container to the app).

Thanks for the effort!

I did not have these issues the past week with commit 7dbcf8658670301d99bff4e9e3b67fc7a7266503.
